src/ServiceControl/Monitoring/MonitoringDataPersister.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace ServiceControl.Monitoring
+{
+    using System.Linq;
+    using Raven.Client;
+    using ServiceControl.Contracts.HeartbeatMonitoring;
+    using ServiceControl.EndpointControl;
+    using ServiceControl.EndpointControl.Contracts;
+    using ServiceControl.Infrastructure;
+    using ServiceControl.Infrastructure.DomainEvents;
+
+    public class MonitoringDataPersister : IDomainHandler<HeartbeatingEndpointDetected>,
+        IDomainHandler<MonitoringEnabledForEndpoint>,
+        IDomainHandler<MonitoringDisabledForEndpoint>
+    {
+
+        public void Handle(HeartbeatingEndpointDetected domainEvent)
+        {
+            var id = DeterministicGuid.MakeId(domainEvent.Endpoint.Name, domainEvent.Endpoint.HostId.ToString());
+
+            using (var session = store.OpenSession())
+            {
+                var knownEndpoint = new KnownEndpoint
```

Comment:
Should we set the session here to use optimistic concurrency so that we don't overwrite previously stored values? I guess it doesn't really make much of a difference though